new-backend/server/apis/v2/services/informative.service.js

```diff
@@ -41,17 +62,29 @@ class InformativeService {
    * @returns
    * @memberof InformativeService
    */
-  async create(documentName, mapName) {
+  async create(documentName, mapName, folderName) {
     try {
       // Add desired file extension to our file's name…
       documentName += ".json";
 
       // …and create a new path to that file.
-      const pathToFile = path.join(
-        process.cwd(),
-        "App_Data/documents",
-        documentName
-      );
+      var pathToFile = "";
+      if (folderName) {
+        pathToFile = path.join(
+          process.cwd(),
+          "App_Data",
+          "documents",
+          folderName,
+          documentName
+        );
+      } else {
+        pathToFile = path.join(
+          process.cwd(),
+          "App_Data",
+          "documents",
+          documentName
+        );
+      }
```

Comment:
You seem to duplicate a lot of code here too.  It's okay to use an empty string in `path.join`, so just ensure that `folderName` isn't `undefined` and refactor this part too.